lib/wasi/src/runtime/module_cache/types.rs

```diff
@@ -93,32 +104,26 @@ impl CacheError {
     }
 }
 
-/// A 256-bit key used for caching.
+/// The 256-bit hash of a WebAssembly module.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
-pub struct Key([u8; 32]);
+pub struct ModuleHash([u8; 32]);
 
-impl Key {
+impl ModuleHash {
     pub fn new(key: [u8; 32]) -> Self {
-        Key(key)
+        ModuleHash(key)
     }
 
-    /// Generate a new [`Key`] based on the SHA-256 hash of some bytes.
-    pub fn sha256(data: impl AsRef<[u8]>) -> Self {
-        let mut hasher = Sha256::default();
-        hasher.update(data);
-        Key::new(hasher.finalize().into())
-    }
+    /// Generate a new [`ModuleCache`] based on the SHA-256 hash of some bytes.
+    pub fn sha256(wasm: impl AsRef<[u8]>) -> Self {
+        let wasm = wasm.as_ref();
+        debug_assert!(
```

Comment:
Initial I wanted to say this could actually be a `Result<(), XXXError>`. 

Only complication is if we wanted to also allow caching of `.wat`, where we should cache the original input rather than a transformed `.wasm`.

But that could also be a separate method. 

===

In any case, I don't think `debug_assert!` is the way to go here: either it's a required invariant, in which case it should be an error, or we just remove it.

Reply:
Removed.